examples/pytorch/language-modeling/run_clm.py

```diff
@@ -543,35 +551,51 @@ def get_mesh(ici_mesh_shape, dcn_mesh_shape=None):
           max_dim = np.argmax(param.shape)
           mesh_shape = [1] * len(param.shape)
           mesh_shape[max_dim] = num_devices
-          print('> [FSDP] Sharding tensor', name, param.shape, mesh_shape)
-          mesh = get_mesh(tuple(mesh_shape))
-          xs.mark_sharding(param, mesh, range(len(param.shape)))
+
+          # Place DCN parallelism on an independent axis
+          dcn_mesh_shape = (model_args.spmd_dcn_parallelism,) + (1,) * len(param.shape)
+          ici_mesh_shape = (1,) + tuple(mesh_shape)
+          mesh = get_mesh(ici_mesh_shape, dcn_mesh_shape)
```

Comment:
what will get_mesh return here?

Reply:
Ah, I need to cut the FSDP dim by `dcn_parallelism`. The result should be `(dcn, n_dev // dcn, 1)` if max_dim = 0

Reply:
hmm, what does this `get_mesh` function and why do we need to pass both `ici` and `dcn` mesh shapes?

Reply:
`get_mesh` allows us to switch between the iota mesh and HybridMesh. HybridMesh requires both `ici` and `dcn` shapes as input, so we need to keep them separate here as well

Reply:
The return is either `xs.Mesh` or `xs.HybridMesh` depending on the `spmd_iota_mesh` flag

Reply:
ah nice